lib/segment/src/index/field_index/full_text_index/mmap_inverted_index/mmap_postings.rs

```diff
@@ -92,36 +91,17 @@ impl MmapPostings {
     /// * `remainder_postings: &'a [PointOffsetType],`
     /// ```
     fn get_reader(&self, header: &PostingListHeader) -> Option<ChunkReader<'_>> {
-        let offset = header.offset as usize;
-        let chunks_len = header.chunks_count as usize;
-        let data_len = header.data_bytes_count as usize;
-        let alignment_len = header.alignment_bytes_count as usize;
-        let remainder_len = header.remainder_count as usize;
-
-        let last_doc_id_bytes = size_of::<PointOffsetType>();
-        let chunks_size_bytes = chunks_len * size_of::<CompressedPostingChunksIndex>();
-        let data_size_bytes = data_len;
-        let alignment_size_bytes = alignment_len;
-        let remainder_size_bytes = remainder_len * size_of::<PointOffsetType>();
-
-        let last_doc_id_offset = offset;
-        let chunks_offset = last_doc_id_offset + last_doc_id_bytes;
-        let data_offset = chunks_offset + chunks_size_bytes;
-        let alignment_offset = data_offset + data_size_bytes;
-        let remainder_offset = alignment_offset + alignment_size_bytes;
-
-        let last_doc_id_mem = self.mmap.get(last_doc_id_offset..chunks_offset)?;
-        let last_doc_id = u32::read_from(last_doc_id_mem)?;
-
-        let chunks_mem = self.mmap.get(chunks_offset..data_offset)?;
-        let chunks = CompressedPostingChunksIndex::slice_from(chunks_mem)?;
-
-        let data = self.mmap.get(data_offset..alignment_offset)?;
-
-        let remainder_mem = self
-            .mmap
-            .get(remainder_offset..remainder_offset + remainder_size_bytes)?;
-        let remainder_postings = u32::slice_from(remainder_mem)?;
+        let bytes = self.mmap.get(header.offset as usize..)?;
+        let (last_doc_id, bytes) = PointOffsetType::read_from_prefix(bytes).ok()?;
+        let (chunks, bytes) = <[CompressedPostingChunksIndex]>::ref_from_prefix_with_elems(
```

Comment:
TBH, `slice_from` in prev version of zerocopy looks easier

Reply:
In short, now they don't provide specialized method for slices, instead you just write `[T]` and use regular methods.

Old (two methods):
- `T::ref_from(…)` for `&T`
- `T::slice_from(…)` for `&[T]`

New (same method):
- `T::ref_from_bytes(…)` for `&T`
- `<[T]>::ref_from_bytes(…)` for `&[T]`

---

As for variations like `ref_from_prefix_with_elems`, they are consistent with the "sequential reading" approach. I.e., from a high level perspective it's seen as this:
```rust
let (some_field_1, bytes) = read_field_1_from_prefix(bytes)?;
let (some_field_2, bytes) = read_field_2_from_prefix(bytes)?;
let (some_field_3, bytes) = read_field_4_from_prefix(bytes, length)?;
let (some_field_4, bytes) = read_field_5_from_prefix(bytes)?;
…
```

It resembles the approach that's often used in parsers, e.g. [this](https://github.com/biomejs/biome/blob/c2dda66cc7533019bbd6c76960fcd68cd94cf1be/xtask/coverage/src/symbols/msts.rs#L244-L317). You don't fiddle with offsets, you read things in a sequence.

---

The name `ref_from_prefix_with_elems` is scary because it supports arbitrary [DST](https://doc.rust-lang.org/nomicon/exotic-sizes.html#dynamically-sized-types-dsts)s, not just slices.